src/output/__tests__/indexer-deduplication.test.ts

```diff
@@ -0,0 +1,179 @@
+import * as TE from "fp-ts/TaskEither";
+import { constVoid, pipe } from "fp-ts/lib/function";
+import { IOutputDocument } from "../elasticsearch/elasticsearch";
+import { IOutputDeduplicationService } from "../elasticsearch/service";
+import { indexerDeduplication } from "../indexer-deduplication";
+
+const mockGet = jest.fn();
+const mockInsert = jest.fn();
+const mockUpdate = jest.fn();
+
+const mockService: IOutputDeduplicationService = {
+  get: mockGet,
+  insert: mockInsert,
+  update: mockUpdate
+};
+
+const indexName = "test_index";
+const document: IOutputDocument = {
+  id: "123",
+  key1: "value1",
+  key2: "value2",
+  _timestamp: 10
+};
+
+describe("indexerDeduplication", () => {
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+  });
+
+  it("indexerDeduplication should insert while retrieving a document (404 error)", async () => {
+    mockGet.mockImplementationOnce(() => TE.left({ statusCode: 404 }));
+    mockInsert.mockImplementationOnce(() => TE.right(constVoid));
+    await pipe(
+      indexerDeduplication(indexName, document)(mockService),
+      TE.bimap(
+        () => {
+          throw new Error(
+            `it should not fail while retrieving a document (404 error)`
+          );
+        },
+        _ => {
+          expect(mockGet).toHaveBeenCalledWith(indexName, document);
+          expect(mockInsert).toHaveBeenCalledWith(indexName, document);
+          expect(mockUpdate).not.toHaveBeenCalled();
+        }
+      )
+    )();
+  });
+
+  it("indexerDeduplication should do nothing while retrieving a document (500 error)", async () => {
```

Comment:
Why? I guess we propagate error on caller if get opertaion doesn't work in order to retrigger transformation